USWDS-Site: Phase two compile updates

amyleadem opened the pull request:
## Summary
Restructured the content in Phase 2 of the Getting Started for Developers guide to add clarity. 

## Related issue
Closes https://github.com/uswds/uswds-site/issues/1708

Recommend releasing this update with the settings page update in PR https://github.com/uswds/uswds-site/pull/1755

## Preview link
Preview link: [Phase 2 Compile ](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/al-add-table-context/documentation/getting-started/developers/phase-two-compile/)

## Problem statement
The guide for compiling USWDS Sass should give clear instructions for both users who want to use `uswds-compile` to create their Sass, and those who want to manage their own Sass compilation. However, the current page creates a single set of steps for both processes. Additionally, the instructions for `uswds-compile` can be confusing if the user is not very familiar with Sass compilation.

## Solution
By separating out the sections and adding clear headers, we can help the user understand which steps they need to take for their project. Additionally, adding more information about setting up the `gulpfile.js` for `uswds-compile` will help give users a clear step-by-step process to get up and running. 

## Testing and review
For copyeditors:
- Proofread for errors and clarity
- Confirm that the page structure makes sense and is easy to follow 

For developers:
- Confirm that all tasks and items listed are accurate and comprehensive
- Confirm that the page structure makes sense and is easy to follow

amyleadem commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -168,29 +219,41 @@ Function | Description
+### Step 7: Verify successful installation
 
-## Step 6: Verify successful installation
-Any time you want to recompile your CSS, run `npx gulp compileSass` from the command line in your project root.
```
Changed this to `npx gulp compile` because that is the name of the task we export in step 2

bonnieAcameron replied:
Smart addition!

amyleadem commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -168,29 +219,41 @@ Function | Description
 
 For any function you defined as an `export` in your `gulpfile.js`, you can run `npx gulp [function]`.
 
-## Step 5: Initialize your project
```
Reorganized this section to move the explanation of what the command does closer to the top.

amyleadem commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -69,62 +88,78 @@ In plain language, this code says:
 
-## Step 4: Create path settings and export compile functions
-Now, create your path settings using the following table:
+### Step 4: Customize path settings
```
Added detail to explain what the path settings are for and examples that show how a user might want to customize

amyleadem commented on pages/documentation/getting-started-developers/phase-two.md:
````diff
@@ -69,62 +88,78 @@ In plain language, this code says:
 exports.init = uswds.init;
 exports.compile = uswds.compile;
-  ```
+exports.watch = uswds.watch;
````
Added the watch command to the example because it is likely a common task people would want in their gulpfile

amyleadem commented on pages/documentation/getting-started-developers/phase-two.md:
````diff
@@ -69,62 +88,78 @@ In plain language, this code says:
+exports.watch = uswds.watch;
+```
+
+### Step 3: Set USWDS version
````
Added this step to highlight the need for setting the USWDS version, as well as describe which values are available.

amyleadem commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -1,45 +1,61 @@
+
+We’ve developed `uswds-compile`, a tool [hosted on GitHub](https://github.com/uswds/uswds-compile), to help teams compile USWDS Sass. This tool provides simple Gulp 4 functions that copy USWDS static assets into your project and transform USWDS Sass into browser-readable CSS. Additionally, `uswds-compile` applies Autoprefixer to all compiled code to make sure your CSS is browser-compatible.
+
+We recommend using `uswds-compile` as a quick way to get up and running with USWDS Sass compilation.
```
Not sure if this actually is the official recommended path for all users, but as a new user (especially one unfamiliar with Sass compilation) I felt like I would want to know if I should choose `uswds-compile`. An explicit recommendation felt like it would help guide a user make a choice.

bonnieAcameron commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -69,62 +88,78 @@ In plain language, this code says:
 - **Build new work on top of that foundation**: Finally, add any custom project styles built from design system code.
 
-After you import the USWDS source code, you can build new styles with USWDS design tokens, functions, and mixins. For the purposes of this guide, we won’t get into custom code, but the important thing to understand is that any custom code should follow the settings and USWDS source code in your Sass entry point.
+  After you load the USWDS source code, you can build new styles with USWDS design tokens, functions, and mixins. For the purposes of this guide, we won’t get into custom code, but the important thing to understand is that any custom code should follow the settings and USWDS source code in your Sass entry point.
```
Line 97: After you've loaded the USWDS source code (provide URL to documentation)

(the tense shift indicates that the step before this one should be completed, not that you need a whole new secret step)

bonnieAcameron commented on pages/documentation/getting-started-developers/phase-two.md:
````diff
@@ -69,62 +88,78 @@ In plain language, this code says:
 
-  {:.site-terminal}
-  ```scss
+- Import the `@uswds/compile` package
+- Set the project's USWDS version
+- Set custom project path settings
+- Export the Gulp functions and/or tasks you need
+
+You can find details about configuring your `gulpfile.js` in steps 3-5.
````
Line 119: Not super clear to me, "in steps 3-5" of which set of steps? Can you name the steps specifically and/or link to them?

bonnieAcameron commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -146,9 +181,25 @@ Setting | Default values - Version 2.x | Default values - Version 3.0 | Descript
+
+### Step 5: Export compile functions
+
+Once your paths are set, export `uswds-compile` functions (summarized in the following table) in your project's `gulpfile.js` to use them in your project.
```
Line 202: "summarized in the Compile Functions Table (hyperlinked to anchor?)"

bonnieAcameron reviewed (changes requested):
This is AMAZING and I think the improvements are really concrete. 

I reviewed this for clarity, plain language (within reason). The main changes I've suggested are making sure to link to relevant documentation when there's a lot to know or unpack behind a seemingly simple step (like "load uswds settings"), but I absolutely defer to your knowledge of their knowledge.

amyleadem commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -1,45 +1,61 @@
+If your project does not yet have a Sass entry point, create a file called `index.scss` or `styles.scss`.
 
-A project often has many Sass files, but typically, there’s a single file that serves as the root — the “homepage” of the Sass — that links out to the others. This root file is also known as the “Sass entry point.” The Sass entry point is the most important stylesheet file in your project because it tells the compiler what source files make up your Sass codebase.
+When your entry point is ready, you'll need to load USWDS Sass. Your Sass entry point will need to do the following three tasks in this order:
```
I have added some language ("For your entry point to complete these tasks") immediately after these tasks are identified to help transition the user. Let me know if you think it needs more. 

I am also wondering if `USWDS Sass` sounds a bit like its own product, rather than just referring to the Sass files inside USWDS. Do you see this too?

amyleadem commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -1,45 +1,61 @@
-3. Include your project’s custom Sass
+1. Define [USWDS settings]({{ site.baseurl }}/documentation/settings/)
+2. Load  USWDS source code
+3. Load your project’s custom Sass
```
It is an interesting idea, for sure. My viewpoint in approaching this is that the  "Getting started for developers" guide is something that anyone familiar with web development at all should be able to complete, regardless of whether they have skill with Sass or not. It would be interesting to define the minimum skillsets we expect out of users though, even if it is for internal use only to start.

amyleadem commented:
@bonnieAcameron made some updates:
1. Added two steps: [Step 3: import uswds-compile package](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/al-add-table-context/documentation/getting-started/developers/phase-two-compile/#step-3-import-the-uswds-compile-package)) and [Step 4: Set USWDS Version](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/al-add-table-context/documentation/getting-started/developers/phase-two-compile/#step-4-set-uswds-version)  so that we create explicit instructions and explanations for each of our tasks listed in [Step 2](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/al-add-table-context/documentation/getting-started/developers/phase-two-compile/#step-2-create-a-gulpfile)
1. Linked each task in the uswds-compile section to its corresponding step
1. Rewrote some pieces of the `About Sass entry points` guide so that each piece is tied to number 1., 2., 3. from the task list
1. Generally just tried to add more clarity throughout. 

Let me know if you have thoughts!

amyleadem commented:
@mejiaj We are working through some restructuring of the Phase 2 compile page. Will you take a look and let us know what you think?

mejiaj commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -1,130 +1,168 @@
+2. **Create the foundation**: Build all USWDS styles from these settings.
 
-- **Create the foundation**: Build all USWDS styles from these settings.
+    Once you have set up your project's custom settings, it is time to pull in the the USWDS source code from the `uswds` package.
```
Extra `the` towards end of sentence.

```suggestion
    Once you have set up your project's custom settings, it is time to pull in the USWDS source code from the `uswds` package.
```

amyleadem replied:
Good catch. Updated in [da0713f](https://github.com/uswds/uswds-site/pull/1766/commits/da0713f8b2fdb0e70227855f62710eedc4d23270)

mejiaj reviewed (changes requested):
LGTM! Only requested change is an extra `the` in a sentence.

mejiaj commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -168,29 +221,41 @@ Function | Description
+
+### Step 7: Initialize your project
+
+When your `gulfile.js` is ready, initialize your project to copy all the necessary Sass, image, font, and Javascript assets from the USWDS source code.
```
Missed the typo here in `gulpfile`.

amyleadem replied:
Thanks. Fixed [da0713f](https://github.com/uswds/uswds-site/pull/1766/commits/da0713f8b2fdb0e70227855f62710eedc4d23270)

mejiaj reviewed (approved):
Looks good, thank you!

bonnieAcameron commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -1,130 +1,168 @@
+## About Sass entry points
 
-## Step 1: Set up your project’s Sass entry point
+A project often has many Sass files, but typically, there’s a single file that serves as the root — the “homepage” of the Sass — that links out to the others. This root file is also known as the “Sass entry point.” The Sass entry point is the most important stylesheet file in your project because it tells the compiler what source files make up your Sass codebase. Often, a project's Sass entry poi ...
```
lovely!

bonnieAcameron commented on pages/documentation/getting-started-developers/phase-two.md:
```diff
@@ -1,130 +1,168 @@
+3. **Build new work on top of that foundation**: Finally, add any custom project styles built from design system code.
 
-After you import the USWDS source code, you can build new styles with USWDS design tokens, functions, and mixins. For the purposes of this guide, we won’t get into custom code, but the important thing to understand is that any custom code should follow the settings and USWDS source code in your Sass entry point.
+    After you've loaded the USWDS source code, you can build new styles with USWDS design tokens, functions, and mixins. For the purposes of this guide, we won’t get into custom code, but the important thing to understand is that any custom code should follow the settings and USWDS source code in your Sass entry point.
```
What if they did want to get into "custom code," is there a resource for that that we could add?
